### DISCUSSION TITLE
Why messagebox load so slowly?

### DISCUSSION THREAD

**Discussion by cuiliang** [HandyOrg/HandyControl#583] [Q&A]
MessageBox showing very slow compared other windows.
Buttons showing like they have slowmothion effects.

I have tried VS timeline, but it didnt show anything slow, render and layout are all fast.

![SlowMessageBox](https://user-images.githubusercontent.com/1972649/102015200-0ea44480-3d95-11eb-9fc3-a0ba3a2f8511.gif)

**Comment by NaBian** [comment]
messagebox 按钮是动态构造的，其它窗口是由xaml定义的，当处在调试模式，这种区别造成的速度差会放大，特别是在实时调试模式启动时区别会更明显。除此之外，3.0中的shell模块是引用的nuget包，其内部报错会影响速度，最新代码shell已经放到仓库中，速度有所提升，但最后还是会比xaml定义的要慢一些。

**Reply by cuiliang** [reply]
我看了一下，按钮的创建在OnApplyTemplate里，在OnLoad之前，这时候窗口还没有显示。
我在quicker里有不少动态创建的控件，速度上和xaml感觉没有可感知的差别。不知道不是和使用了template有关，quicker里的窗口没有使用template。
等有空再研究研究。
